src/JsonSerialisation.ts

```diff
@@ -48,16 +41,214 @@ export class JsonSerialisation{
 		return list;
 	}
 
-	public static DictionaryRuntimeObjsToJObject(dictionary: Map<string, InkObject>){
-		let jsonObj: JObject = {};
-
+	public static WriteDictionaryRuntimeObjs(writer: SimpleJson.Writer, dictionary: Map<string, InkObject>){
+		writer.WriteObjectStart();
 		for (let [key, value] of dictionary){
-			let runtimeObj = asOrNull(value, InkObject);
-			if (runtimeObj != null)
-				jsonObj[key] = this.RuntimeObjectToJToken(runtimeObj);
+			writer.WritePropertyStart(key);
+			this.WriteRuntimeObject(writer, value);
+			writer.WritePropertyEnd();
+		}
+		writer.WriteObjectEnd();
+	}
+
+	public static WriteListRuntimeObjs(writer: SimpleJson.Writer, list: InkObject[]){
+		writer.WriteArrayStart();
+		for (let value of list){
+			this.WriteRuntimeObject(writer, value);
+		}
+		writer.WriteArrayEnd();
+	}
+
+	public static WriteIntDictionary(writer: SimpleJson.Writer, dict: Map<string, number>){
+		writer.WriteObjectStart();
+		for (let [key, value] of dict){
+			writer.WriteIntProperty(key, value);
+		}
+		writer.WriteObjectEnd();
+	}
+
+	public static WriteRuntimeObject(writer: SimpleJson.Writer, obj: InkObject): void{
+		// var container = obj as Container;
+		let container = asOrNull(obj, Container);
+		if (container) {
+			return this.WriteRuntimeContainer(writer, container);
+		}
+
+		// var divert = obj as Divert;
+		let divert = asOrNull(obj, Divert);
+		if (divert) {
+			let divTypeKey = '->';
+			if (divert.isExternal)
+				divTypeKey = 'x()';
+			else if (divert.pushesToStack) {
+				if (divert.stackPushType == PushPopType.Function)
+					divTypeKey = 'f()';
+				else if (divert.stackPushType == PushPopType.Tunnel)
+					divTypeKey = '->t->';
+			}
+
+			let targetStr;
+			if (divert.hasVariableTarget)
+				targetStr = divert.variableDivertName;
+			else
+				targetStr = divert.targetPathString;
+
+			writer.WriteObjectStart();
+			writer.WriteProperty(divTypeKey, targetStr);
+
+			if (divert.hasVariableTarget)
+				writer.WriteProperty('var', true);
+
+			if (divert.isConditional)
+				writer.WriteProperty('c', true);
+
+			if (divert.externalArgs > 0)
+				writer.WriteIntProperty('exArgs', divert.externalArgs);
+
+			writer.WriteObjectEnd();
+		}
+
+		// var choicePoint = obj as ChoicePoint;
+		let choicePoint = asOrNull(obj, ChoicePoint);
+		if (choicePoint) {
+			writer.WriteObjectStart();
+			writer.WriteProperty('*', choicePoint.pathStringOnChoice);
+			writer.WriteIntProperty('flg', choicePoint.flags);
+			writer.WriteObjectEnd();
```

Comment:
Same here and in the next blocks where we appear to be missing return statements. Unless they aren't there for a particular reason?

Reply:
I have no idea. Must have been half asleep, very good catch thanks. I'll add them back.